### PR TITLE
LPS-168928 Create ConfigurationScreen for viewing and modifying Feature Flags

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-168928](https://issues.liferay.com/browse/LPS-168928).

### CODE REVIEW

**Comment by drewbrokke** [comment]
@pei-jung @david-truong FYI

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by melodywu** [comment]
Hello @drewbrokke , There are two related failures found in ci:test:relevant result, 
Issue 1: [com.liferay.feature.flag.web.internal.helper.FeatureFlagPropsHelperTest](https://testray.liferay.com/zh/home/-/testray/case_results/2104423707?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcases%2F2104423701%2Fview%3Fcur%3D1%26delta%3D200%26testrayCaseId%3D2104423701%26orderByCol%3DcreateDate_sortable%26orderByType%3Ddesc)
{code}
[exec] com.liferay.feature.flag.web.internal.model.PreferenceAwareFeatureFlagTest > initializationError STARTED
     [exec] 
     [exec] com.liferay.feature.flag.web.internal.model.PreferenceAwareFeatureFlagTest > initializationError FAILED
     [exec]     java.lang.Exception: Method setUp() should be public
     [exec]         at org.junit.runners.model.FrameworkMethod.validatePublicVoid(FrameworkMethod.java:96)
     [exec]         at org.junit.runners.model.FrameworkMethod.validatePublicVoidNoArg(FrameworkMethod.java:74)
     [exec]         at org.junit.runners.ParentRunner.validatePublicVoidNoArgMethods(ParentRunner.java:155)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.validateInstanceMethods(BlockJUnit4ClassRunner.java:187)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.collectInitializationErrors(BlockJUnit4ClassRunner.java:128)
     [exec]         at org.junit.runners.ParentRunner.validate(ParentRunner.java:416)
     [exec]         at org.junit.runners.ParentRunner.<init>(ParentRunner.java:84)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.<init>(BlockJUnit4ClassRunner.java:65)
     [exec]         at org.junit.internal.builders.JUnit4Builder.runnerForClass(JUnit4Builder.java:10)
     [exec]         at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
     [exec]         at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
     [exec]         at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.j
{code}

Issue2: [com.liferay.feature.flag.web.internal.model.PreferenceAwareFeatureFlagTest](https://testray.liferay.com/home/-/testray/case_results/2104423738)
{code}
exec] 
     [exec] com.liferay.feature.flag.web.internal.model.PreferenceAwareFeatureFlagTest > initializationError FAILED
     [exec]     java.lang.Exception: Method setUp() should be public
     [exec]         at org.junit.runners.model.FrameworkMethod.validatePublicVoid(FrameworkMethod.java:96)
     [exec]         at org.junit.runners.model.FrameworkMethod.validatePublicVoidNoArg(FrameworkMethod.java:74)
     [exec]         at org.junit.runners.ParentRunner.validatePublicVoidNoArgMethods(ParentRunner.java:155)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.validateInstanceMethods(BlockJUnit4ClassRunner.java:187)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.collectInitializationErrors(BlockJUnit4ClassRunner.java:128)
     [exec]         at org.junit.runners.ParentRunner.validate(ParentRunner.java:416)
     [exec]         at org.junit.runners.ParentRunner.<init>(ParentRunner.java:84)
     [exec]         at org.junit.runners.BlockJUnit4ClassRunner.<init>(BlockJUnit4ClassRunner.java:65)
     [exec]         at org.junit.internal.builders.JUnit4Builder.runnerForClass(JUnit4Builder.java:10)
     [exec]         at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
     [exec]         at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
     [exec]         at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
     [exec]         at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:33)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:78)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
     [exec]         at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
     [exec]         at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]         at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]         at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]         at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]         at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
{code}
Could you help with confirm @drewbrokke ? Thanks in advance.I'd trigger the CI to build a bundle to confirm the issue on local as well.

**Comment by Sharryxue** [comment]
Specific issues found in the ci:test:relevant result. Included the failures above mentioned by Melody. Could you please take care of the issues @drewbrokke, thanks in advance.
- [modules-compile-jdk8](https://testray.liferay.com/home/-/testray/case_results/2104420235), refer the [console log](https://testray.liferay.com/reports/production/logs/2022-12/test-1-37/test-portal-acceptance-pullrequest(master)/6780/modules-compile-jdk8/0/3/jenkins-console.txt.gz?authuser=0), [portal-license-smoke-jdk8](https://testray.liferay.com/home/-/testray/case_results/2104423653), refer the [console log](https://testray.liferay.com/reports/production/logs/2022-12/test-1-37/test-portal-acceptance-pullrequest(master)/6780/portal-license-smoke-jdk8/0/0/jenkins-console.txt.gz?authuser=0).
```
     [exec] FAILURE: Build failed with an exception.
     [exec] 
     [exec] * What went wrong:
     [exec] Execution failed for task ':apps:feature-flag:feature-flag-web:generateJSPJava'.
     [exec] > file:/opt/dev/projects/github/liferay-portal/modules/apps/feature-flag/feature-flag-web/classes/META-INF/resources/init.jsp(20,57) PWC6188: The absolute uri: http://liferay.com/tld/react cannot be resolved in either web.xml or the jar files deployed with this application
     [exec] 
     [exec] * Try:
     [exec] Run with --ifo or --debug option to get more log output. Run with --scan to get full insights.
     [exec] 
     [exec] * Exception is:
     [exec] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':apps:feature-flag:feature-flag-web:generateJSPJava'.
```
- [PreferenceAwareFeatureFlagTest](https://testray.liferay.com/home/-/testray/case_results/2104423738), full [console log](https://testray.liferay.com/reports/production/logs/2022-12/test-1-37/test-portal-acceptance-pullrequest(master)/6780/modules-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0).
`initializationError: java.lang.Exception: Method setUp() should be public`
- [FeatureFlagPropsHelperTest](https://testray.liferay.com/zh/home/-/testray/case_results/2104423707), full [console log](https://testray.liferay.com/reports/production/logs/2022-12/test-1-37/test-portal-acceptance-pullrequest(master)/6780/modules-unit-jdk8/0/0/jenkins-console.txt.gz?authuser=0).
`testGetTitle: org.junit.ComparisonFailure: expected:<[]> but was:<[XYZ-123]>`

**Comment by drewbrokke** [comment]
ci:test:relevant

Just pushed fixes for the failures.

**Comment by drewbrokke** [comment]
ci:test:sf

**Comment by drewbrokke** [comment]
ci:test:relevant

**Comment by drewbrokke** [comment]
ci:reevaluate:1267401_6262

File: modules/apps/feature-flag/feature-flag-web/src/test/java/com/liferay/feature/flag/web/internal/util/FeatureFlagJSONUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ */
+public class FeatureFlagJSONUtilTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Test
+	public void testToJSONString() throws Exception {
```
Comment (drewbrokke) [inline modules/apps/feature-flag/feature-flag-web/src/test/java/com/liferay/feature/flag/web/internal/util/FeatureFlagJSONUtilTest.java:43]: - [ ] Fix test name
- [ ] Fix commit message typo

**Comment by melodywu** [comment]
No related issue found in ci:test:relevant. Note: [modules-compile-jdk8](https://testray.liferay.com/home/-/testray/case_results/2104420235) , [PreferenceAwareFeatureFlagTest](https://testray.liferay.com/home/-/testray/case_results/2104423738), and [FeatureFlagPropsHelperTest](https://testray.liferay.com/zh/home/-/testray/case_results/2104423707) all passed.

**Comment by melodywu** [comment]
No related issue found in ci:test:relevant.

**Comment by drewbrokke** [comment]
@Sharryxue @melodywu if this one seems ok to you, then I think we're good to forward it.

**Comment by Sharryxue** [comment]
Manually forwarded. Refer https://github.com/brianchandotcom/liferay-portal/pull/127426.

**Comment by Sharryxue** [comment]
ci:close

**Pull request closed without merging**